clab/dependency_manager.go

```diff
@@ -34,19 +36,25 @@ type defaultDependencyManager struct {
 	// Names of the nodes that depend on a given node are listed here.
 	// On successful creation of the said node, all the depending nodes (dependers) wait groups will be decremented.
 	nodeDependers map[string][]string
+
+	nodeCreated map[string]*sync.WaitGroup
```

Comment:
@steiler do we need a WaitGroup here? I have the following concerns

1. the `nodeCreated` name may be not a great one here, since it says that the node is either created or not. So we don't really say if it running, for example.  
2. For the future impl I think it makes sense to use the `nodeStatus` field with an enumerated string field such as `created`, `running`, `exited`, etc.
3. But if we just want to start small here and track running status, i'd propose to name this as `nodeStatus` and just operate with `""` and `"running"` string values.  
When initialized, the empty string means node status is unknown, and when the node has been running we set it to `running`

Reply:
Scratch that, I see that in the WaitNode we rely on `wg.Wait()` which makes sense

Reply:
so maybe just to clarify here, does dm finishes when it creates the node or when the node is running?